core/src/main/scala/cats/data/RWST.scala

```diff
@@ -0,0 +1,550 @@
+package cats
+package data
+
+import cats.functor.{ Contravariant, Bifunctor, Profunctor }
+import cats.syntax.either._
+
+/**
+  * Represents a stateful computation in a context `F[_]`, over state `S`, with an initial environment `E`, an accumulated log `L` and a result `A`.
+  *
+  * In other words, it is a pre-baked stack of `[[ReaderT]][F, E, A]`, `[[WriterT]][F, L, A]`
+  * and `[[StateT]][F, S, A]`.
+  */
+final class RWST[F[_], E, S, L, A](val runF: F[(E, S) => F[(L, S, A)]]) extends Serializable {
```

Comment:
Would it be better to have `ReaderWriterStateT` be the canonical name, with `RWST` as an alias?  I look at this as being similar to `FunctionK` and `~>`.